addons/mail/models/mail_thread.py

```diff
@@ -702,7 +702,7 @@ def _routing_create_bounce_email(self, email_from, body_html, message, **mail_va
         bounce_from = self.env['ir.mail_server']._get_default_bounce_address()
         if bounce_from:
             bounce_mail_values['email_from'] = tools.formataddr(('MAILER-DAEMON', bounce_from))
-        else:
+        elif self.env['ir.config_parameter'].sudo().get_param("mail.catchall.alias") not in message['To']:
```

Comment:
The `else` was introduced by 1838191eece59051805862ed6ce914bf5e5d1efb, and it's not very clear why. Perhaps just for tests?

I find it too bad that this PR will re-introduce leaks of ICP params here, when the logic was nicely encapsulated in `ir.mail_server._get_default_bounce_address()`.

When does the problem occur? When you have not set a `mail.catchall.domain` (only case where `_get_default_bounce_address()` will return `None`) but still happen to receive a direct email to the catchall alias? That doesn't look like it could happen in a normal situation, only for pure dev/test. Not sure it's worth a patch.

Also note that removing the original `else` also solves the problem, and it's not clear that we need it? @tde-banana-odoo  can certainly clarify ;-)

Reply:
If I remember well ... I forced author_id to be False to avoid having robodoo everywhere, then we did not have any email_from in some cases.

We could fallback on current user's email as before the TdeCleanYourBrol commit, as we are quite sure in real prod environment we will have a bounce_from (+ a small comment to tell it).

Reply:
That alternative sounds good, and thanks for the explanation :)

Reply:
On my side, the self.env.user.email_normalized is `odoobot@example.com`, are we sure we want to use it ? (my email provided rejected it)

Reply:
Looks like the expected behavior indeed, if you have not defined a proper `mail.catchall.domain`, no? If you were in a real deployment you'd either have a domain configured, or you would not receive emails to the catchall alias.

It's also the first step when configuring email integration (and is auto-filled on our SaaS):

![screenshot-2609567-saas-13-2-dce0c2 runbot37 odoo com-2020 05](https://user-images.githubusercontent.com/2039870/81723885-cbdb6080-9483-11ea-9572-bbeb4f092fc5.png)